java/code/src/com/redhat/rhn/common/db/datasource/xml/SystemReport_queries.xml

```diff
@@ -83,34 +83,53 @@
                     INNER JOIN rhnservergroup ON rhnservergroupmembers.server_group_id = rhnservergroup.id
                     INNER JOIN web_customer ON rhnservergroup.org_id = web_customer.id
            WHERE rhnservergroup.group_type IS NULL
-          order by server_id, server_group_id offset :offset limit :limit
+        ORDER BY server_id, server_group_id OFFSET :offset LIMIT :limit
   </query>
 </mode>
 
-<mode name="SystemPrimaryAddress" class="">
+<mode name="SystemNetInterface" class="">
     <query params="offset, limit">
           SELECT server_id AS system_id
-                    , rsna4.address AS ip4_addr
-                    , rsna4.netmask AS ip4_netmask
-                    , rsna4.broadcast AS ip4_broadcast
-                    , rsna6.address AS ip6_addr
-                    , rsna6.netmask AS ip6_netmask
-                    , rsna6.scope AS ip6_scope
-            FROM rhnservernetinterface rsni
-                    LEFT JOIN rhnservernetaddress4 rsna4 ON rsni.id = rsna4.interface_id
-                    LEFT JOIN rhnservernetaddress6 rsna6 ON rsni.id = rsna6.interface_id
-           WHERE id IN ( SELECT min(id)
-                           FROM rhnservernetinterface
-                          WHERE is_primary = 'Y'
-                       GROUP BY server_id
-                 )
-          order by server_id offset :offset limit :limit
-  </query>
+                    , id AS interface_id
+                    , name
+                    , hw_addr AS hardware_address
+                    , module
+                    , COALESCE(is_primary, 'N') = 'Y' AS primary_interface
+            FROM rhnservernetinterface
+        ORDER BY system_id, interface_id OFFSET :offset LIMIT :limit
+    </query>
+</mode>
+
+<mode name="SystemNetAddressV4" class="">
+    <query params="offset, limit">
+          SELECT rhnservernetinterface.server_id AS system_id
+                    , rhnservernetinterface.id AS interface_id
+                    , rhnservernetaddress4.address
+                    , rhnservernetaddress4.netmask
+                    , rhnservernetaddress4.broadcast
+            FROM rhnservernetinterface
+                    INNER JOIN rhnservernetaddress4 ON rhnservernetinterface.id = rhnservernetaddress4.interface_id
+        order by system_id, interface_id, address OFFSET :offset LIMIT :limit
```

Comment:
you uppercased keywords for other queries, so I think you should do it here as well.

Reply:
Fixed